src/components/subscribe/subscribe.test.ts

```diff
@@ -0,0 +1,273 @@
+import path from 'path';
+import type { PleasantestUtils } from 'pleasantest';
+import { withBrowser, getAccessibilityTree } from 'pleasantest';
+import { loadTwigTemplate, loadGlobalCSS } from '../../../test-utils';
+
+// Helper to load the Twig template file
+const componentMarkup = loadTwigTemplate(
+  path.join(__dirname, './subscribe.twig')
+);
+// Helper to load the demo Twig template file
+const demoMarkup = loadTwigTemplate(path.join(__dirname, './demo/demo.twig'));
+
+// Helper to initialize the component JS
+const initJS = (utils: PleasantestUtils) =>
+  utils.runJS(`
+    import { initSubscribe } from './subscribe'
+    export default () => initSubscribe(
+      document.querySelector('.js-subscribe')
+    )
+  `);
+
+describe('Subscription', () => {
+  test(
+    'should use semantic markup',
+    withBrowser(async ({ utils, page }) => {
+      await loadGlobalCSS(utils);
+      await utils.loadCSS('./subscribe.scss');
+      await utils.injectHTML(await componentMarkup());
+      await initJS(utils);
+
+      expect(await getAccessibilityTree(page)).toMatchInlineSnapshot(`
+        document
+          heading "Never miss an article!"
+            text "Never miss an article!"
+          status
+            text "Notifications have been turned off."
+          button "Get notifications"
+          link "Get Weekly Digests"
+            text "Get Weekly Digests"
+          form "Get Weekly Digests"
+            heading "Get Weekly Digests"
+              text "Get Weekly Digests"
+            text "Email"
+            textbox
```

Comment:
I switched to using the Input Group  with an Input component and this value changed. Before when I was using a vanilla `<input>`, it read `textbox "Email"`. Now it only reads `textbox`. The "name" value seems to have gone away?

I reviewed both the vanilla and Input Group with the Input component and I was unable to notice a difference in the markup so I'm not sure why this changed. 🤔

Reply:
@gerardo-rodriguez I tried to debug this locally, I logged the markup using:

```js
console.log(await page.evaluate(() => document.body.innerHTML));
```

right before the snapshot.

It looks like the `id`/`for` attributes aren't getting set correctly.

```html
<label class="c-subscribe__form-input-label" for="subscription-email-">
          Email
        </label>
        <div class="o-input-group c-subscribe__form-input-group">


      <input type="email" class="c-input c-subscribe__form-input" id="subscription-email-null" placeholder="Your
 Email Address" autocomplete="home email" required="">
```

Looks like we are missing the `form_id` prop?

Reply:
> I logged the markup using

Great idea! We should add this “recipe” tip somewhere in the Pleasantest docs. It wasn’t obvious I could do that. LOL

> Looks like we are missing the form_id prop?

Oh, odd! Let’s create a GH issue for this. Can you help me with that? If not, no worries, I can do so later this morning. 

Thanks for following up, @calebeby!

Reply:
https://github.com/cloudfour/cloudfour.com-patterns/pull/1689